src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```diff
@@ -0,0 +1,135 @@
+ */
+
+@RestController
+@RequestMapping(path = "/api/bookings")
+public class BookingsController {
+    private static final ResponseEntity<Object> SUCCESS_RESPONSE = new ResponseEntity<>(HttpStatus.OK);
+    private static final ResponseEntity<Object> FAILURE_RESPONSE = new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+    private static final Logger LOGGER = Logger.getLogger(BookingsController.class.getName());
+    private final BookingService bookingService;
+
+    public BookingsController(BookingService bookingService) {
+        this.bookingService = bookingService;
+    }
+
+    @PostMapping
+    @ResponseStatus(value = HttpStatus.CREATED)
+    public ResponseEntity<Object> createBooking(@RequestBody final Booking booking) {
+        try {
+            LOGGER.info("Booking created successfully. Update time: " + bookingService.createBooking(booking));
+            return SUCCESS_RESPONSE;
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return FAILURE_RESPONSE;
+        }
+    }
+
+    @GetMapping("/getBookingsByUserId/{userId}")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getBookingsByUserId(@PathVariable(value = "userId") UUID userId) {
+        LinkedHashSet<Booking> bookingsByUserId = new LinkedHashSet<>();
+        try {
+            bookingsByUserId = bookingService.getBookings(userId);
+            LOGGER.info("GET BOOKINGS BY userID - API endpoint invoked");
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getBooking/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> getBooking(@PathVariable(value = "id") final UUID id) {
+        Booking booking = null;
+        try {
+            booking = bookingService.getBookingByID(id);
+            LOGGER.info("GET BOOKING BY ID - API endpoint invoked");
+            return new ResponseEntity<>(booking, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(booking, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/showAllBookings")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getAllBookings() {
+        LinkedHashSet<Booking> bookings = new LinkedHashSet<>();
+        try {
+            bookings = bookingService.getAllBookings();
+            return new ResponseEntity<>(bookings, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(bookings, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/delete")
```

Comment:
Could you highlight why you have used put and not delete/post? Also, should this method be annotated with `@ResponseBody`?